File: C++/1400. Construct K Palindrome Strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        if(s.size()<k)
        return false;
        map<char,int>m;
        for(char c:s)
        m[c]++;
        int c=0;
        for(auto i=m.begin();i!=m.end();i++){
            if(i->second%2==1)
            c++;
            if(c>k)
            return false;
        }
        return true;

    }
};

File: JAVA/1400. Construct K Palindrome Strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        if(s.length()<k)
        return false;
        int[] f=new int[26];
        for(int i=0;i<s.length();i++){
            f[s.charAt(i)-'a']++;
        }
        int c=0;
        for(int i=0;i<26;i++){
            if(f[i]==0)
            continue;
            if(f[i]%2==1)
            c++;
            if(c>k)
            return false;
        }
        return true;
    }
}